class.jetpack-cli.php

```diff
@@ -1476,6 +1476,107 @@ private function partner_provision_error( $error ) {
 		) ) );
 		exit( 1 );
 	}
+
+	/**
+	 * Creates the essential files in Jetpack to start building a Gutenberg block.
+	 *
+	 * All files will be created in a directory under extensions/blocks named based on the block title or a specific given slug.
+	 *
+	 * ## OPTIONS
+	 *
+	 * title : Block name, also used to create the slug. If it's something like "Logo gallery", the slug will be 'logo-gallery'
+	 * --slug: Specific slug to identify the block that overrides the one generated base don the title.
+	 * --description: Allows to provide a text description of the block.
+	 * --keywords: Provide up to three keywords separated by comma so users  when they search for a block in the editor.
+	 * --external-edit: Creates an additional file to create the edit as an external component. It doesn't require any value.
+	 *
+	 * ## EXAMPLES
+	 *
+	 * wp jetpack block "Cool Block"
+	 * wp jetpack block "Amazing Rock" --slug="good-music" --description="Rock the best music on your site"
+	 * wp jetpack block "Jukebox" --keywords="music, audio, media"
+	 *
+	 * @subcommand block
+	 * @synopsis <title> [--slug] [--description] [--keywords] [--external-edit]
+	 */
+	public function block( $args, $assoc_args ) {
+		// It's ok not to check because if it's set, WPCLI exits earlier.
+		$title = $args[0];
+
+		$slug = isset( $assoc_args['slug'] )
+			? $assoc_args['slug']
+			: sanitize_title( $title );
+
+		if ( ! preg_match( '/^[a-z][a-z0-9\-]*$/', $slug ) ) {
+			WP_CLI::error( esc_html__( 'Invalid block slug. They can contain only lowercase alphanumeric characters or dashes, and start with a letter', 'jetpack' ) . ' 👻' );
+		}
+
+		global $wp_filesystem;
+		if ( ! WP_Filesystem() ) {
+			WP_CLI::error( esc_html__( "Can't write files", 'jetpack' ) . ' 😱' );
+		}
+
+		$path = JETPACK__PLUGIN_DIR . "extensions/blocks/$slug";
+
+		$wp_filesystem->mkdir( $path );
```

Comment:
This isn't handling if you put in a title/slug that would conflict with an existing one, e.g. `wp jetpack block GiF` will partly overwrite the existing gif block.

Reply:
Good catch